app/jobs/cache/browser_channels/prefix_list.rb

```diff
@@ -0,0 +1,85 @@
+begin
+  # Have to throw in a begin rescue block otherwise
+  # Zeitwerk::NameError (expected file $DIR/protos/channel_responses.rb to define constant ChannelResponses, but didn't)
+  # gets thrown.
+  require './protos/publisher_list'
+rescue
+end
+begin
+  require './protos/channel_responses'
+rescue
+end
+
+class Cache::BrowserChannels::PrefixList
+  include Sidekiq::Worker
+  sidekiq_options queue: :low, retry: false
+
+  # Might need to adjust the value based on 
+  PREFIX_LENGTH = 9
+
+  def perform
+    # We only care about first PREFIX_LENGTH number of nibbles of the prefix.
+    # Don't waste memory reading excess
+    temp_file = save_main_file!
+    save_to_s3!(temp_file_path: temp_file.path, save_to_filename: "prefixes")
+    cleanup!(temp_file_path: temp_file)
+
+    temp_file = save_differential_file!
+    date = Date.yesterday.strftime("%Y-%m-%d")
+    save_to_s3!(temp_file_path: temp_file.path, save_to_filename: "prefixes-#{date}")
+    cleanup!(temp_file_path: temp_file)
+  end
+
+  def save_main_file!
+    result = ActiveRecord::Base.connection.execute("
+        SELECT SUBSTRING(sha2_base16, 1, #{PREFIX_LENGTH})
+        FROM site_banner_lookups
+        WHERE wallet_status != #{PublishersPb::WalletConnectedState::NO_VERIFICATION}"
+        ).map { |r| r['substring'] }.to_json
```

Comment:
Instead of JSON, we need to sort the prefixes and then hex decode them so that we have a raw byte string. I think this can be done in Ruby like so:

```rb
# For an array of hex prefixes |hex_values|
raw_values = hex_values.sort().map { |item| [item].pack('H*') }

# Joining the array elements into one long raw string
raw_data = raw_values.join()
```

Reply:
Sorting is resolved.